R/prepare_GISTIC_inputs.R

```diff
@@ -0,0 +1,234 @@
+      }
+    }
+    # get the seg_data
+    seg_for_gistic <- get_cn_segments(
+        these_samples_metadata = these_samples_metadata,
+        flavour = flavour,
+        fill_missing_with = "avg_ploidy",
+        projection = projection
+    )
+    #check how many samples we have data from
+    u_id = length(unique(seg_for_gistic$ID))
+    print(paste("got data from",u_id,"sample IDs"))
+    seg_for_gistic <- seg_for_gistic %>%
+        dplyr::select(ID:log.ratio) %>%
+        mutate(LOH_flag = ifelse(LOH_flag == 99, 0, LOH_flag))
+    print("checking for overlapping segments")
+    seg_for_gistic <- check_overlap(seg_for_gistic)
+
+    if ("overlap" %in% seg_for_gistic$overlap_status) {
+        print("resolving overlapping segments")
+        seg_for_gistic <- solve_overlap(seg_for_gistic)
+    } else {
+      seg_for_gistic <- seg_for_gistic %>%
+            select(-overlap_status, -region_size)
+    }
+
+    # write to disk and make markers file
+    write_tsv(seg_for_gistic, file = out_seg)
+    temp_markers <- paste0(output_dir, "temp_markers")
+    print("processing for GISTIC")
+    cmd <- paste("sed '1d'", out_seg, "| cut -f2,3 > ", temp_markers)
+    system(cmd)
+    cmd <- paste("sed '1d'", out_seg, "| cut -f2,4 >> ", temp_markers)
+    system(cmd)
+    cmd <- paste("sort -V -k1,1 -k2,2nr", temp_markers, "|uniq | nl > ", markers)
+    system(cmd)
+    # sed '1d' {input.seg} | cut -f2,3 > {output.temp_markers}
+    # sed '1d' {input.seg} | cut -f2,4 >> {output.temp_markers}
+    # sort -V -k1,1 -k2,2nr {output.temp_markers} | uniq | nl > {output.markers}
+    print("done!")
+    print("Running gistic:")
+    print("conda activate /projects/rmorin_scratch/conda_environments/30ee048690c736b4964ed302c88be45f_")
+    print("gistic2 -b /projects/rmorin/projects/gambl-repos/gambl-rmorin/gistic_test")
+    print("-seg ./inputs/all.seg -mk inputs/markers -refgene ref/grch37.refgene.mat -genegistic 1")
+    print("-broad 1 -savegene 1 -conf 0.9 -v 30 -saveseg 0 -savedata 0")
+
+}
+
+
+check_overlap <- function(seg) {
+    highest_end <- 0
+    overlap <- c()
+    region_sizes <- c()
+    for (i in 1:nrow(seg)) {
+        if (i > 1 && seg$ID[i] == seg$ID[i - 1] && seg$chrom[i] == seg$chrom[i - 1]) {
+            if (seg$start[i] >= highest_end) {
+                overlap[i] <- "NOToverlap"
+                region_sizes[i] <- "FALSE"
+            } else {
+                overlap[i] <- "overlap"
+                region_sizes[i] <- (seg$end[i] - seg$start[i])
+            }
+            if (seg$end[i] > highest_end) {
+                highest_end <- seg$end[i]
+            }
+        } else {
+            highest_end <- seg$end[i]
+            overlap[i] <- "NA"
+            region_sizes[i] <- (seg$end[i] - seg$start[i])
+        }
+    }
+    seg <- seg %>% mutate(overlap_status = overlap, region_size = region_sizes)
+    return(seg)
+}
+
+solve_overlap <- function(seg) {
```

Comment:
this reminds me of the first iteration of the fill_segments script that was done in python :sweat_smile: I will just trust that this works and handles all edge cases!